tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/AbstractPortalBundle.java

```diff
@@ -404,4 +408,58 @@ else if( configType.equals( CONFIG_TYPE_SERVER ) )
             }
         }
     }
+
+    protected void setHttpPortValue(
```

Comment:
This code should not be shared between all portal bundles.  This code should be in specific tomcat and jboss.  even if the xml files are very similar, this implementation should be in separate places.